src/lib/geo/geo.c

```diff
@@ -298,6 +298,43 @@ __EXPORT float get_distance_to_next_waypoint(double lat_now, double lon_now, dou
 	return CONSTANTS_RADIUS_OF_EARTH * c;
 }
 
+__EXPORT void create_waypoint_from_line_and_dist(double lat_A, double lon_A, double lat_B, double lon_B, float dist,
+		double *lat_target, double *lon_target)
+{
+	float heading;
+
+	if (fabsf(dist) < FLT_EPSILON) {
+		*lat_target = lat_A;
```

Comment:
heading is undefined in here.
I would move "float heading;" into the else if / else cases.